### PR TITLE
expose additional context when rendering

### PR DESCRIPTION
**Pull request by erikmueller**
Inspired by http://blog.overstuffedgorilla.com/render-react-with-phoenix/ my use case was playing around with rendering react-router v4 inside an elixir application. 

I had the problem of not being able to pass `context` that is set during route matching in the react-router (and maybe populated with a response status) back to the `plug.Router` (to set the statusCode). 

I solved this with setting a `context` object on the component itself and passing that back to the caller (the `plug.Router` in that case). This may not be the most elegant solution but the only that came to my mind. Maybe s/o has something more awesome up their sleeves ;)

Basically works like described in the `contextComponent` and sInce I was too afraid to break sth I wrote some tests ^^

### CODE REVIEW

**Comment by erikmueller** [comment]
@mjackson any thoughts?

**Comment by mjackson** [comment]
Hey @erikmueller, sorry for the long silence here. I like this solution! Also, thanks for the tests! 😅

I'll merge and release in the next patch release.
